### ISSUE TITLE
implement realtime extension support: AAPXS SysEx8

### ISSUE THREAD

**Issue by atsushieno** [atsushieno/aap-core#169]
As it has been written in several issues and documents (e.g. issue #73, issue #145, issue #158), current Binder-based `extension()` support does not cover realtime usage scenario and they should be covered as in MIDI2 system exclusive messages at `process()`.

Now that we have refresh implementation for plugin manager with `aap::MidiDestinationNode` which is ready to be used in `aap::PluginPlayer`, it is time to actually implement it.

There are handful of blocks that need new code:

- [x] `PluginPlayer needs to provide users ways to handle its MIDI2 outputs, either via event handler or provide direct buffer to MIDI outputs.
- [x] cmidi2 needs sysex8 data extraction feature
- [x] implement MIDI2 AAPXS SysEx8 parser and generator
- [x] MIDI AAPXS service needs to parse MIDI2 sysex8 inputs into extension invocation, serialized as extension bytes.
- [x] `aap::LocalPluginInstance` needs to integrate this MIDI input parsing support.
- [x] MIDI AAPXS client needs to generate MIDI inputs from extension invocation, serialized as extension bytes. -> post-0.7.8 task
- [x] `aap::RemotePluginInstance` needs to integrate this MIDI output generation support.
- [x] the new plugin manager needs to implement realtime extension invocation. MIDI Program Change is a good example for this (involves preset setter).
- [x] There should be asynchronous callback support in every async-ified (non-RT) extension function, and they need to be invoked at each AAPXS implementation.
- [x] C++ wrappers in `standard-extensions.h` and its implementation also needs async-ification.

Unless it is going to be too complicated, it should be done before 0.7.8 release.

**Comment by atsushieno** [comment]
> Unless it is going to be too complicated, it should be done before 0.7.8 release.

As issues #171 is post-0.7.8 work, the entire work is inherently moved to post-0.7.8 as well. Until then, we will have proof-of-concept setPresetIndex() call at `aap::AudioPluginNode::setPresetIndex(int index)` to conditionally generate AAPXS SysEx8 and send it.

**Comment by atsushieno** [comment]
Most of the work is now done in `aapxs-v080` branch, but I noticed that `binder-client-as-plugin`'s `get_extension()` requires public extension API implementation for each AAPXS. It also needs to be provided by each AAPXS developer.

It is possible to achieve that by `TypedAAPXS` implementation classes, but that requires `AAPXSDispatcher` and its instantiated shared memory buffers, while `AAPXSDefinition` itself is basically all statically resolved. We likely need to provide another way to instantiate the registry, kind of.

**Comment by atsushieno** [comment]
^ is resolved as `get_plugin_extension_proxy()` member function in `AAPXSDefinition`.

**Comment by atsushieno** [comment]
We can close this issue once `aapxs-v080` branch is merged to `main`.

**Comment by atsushieno** [comment]
It is now merged into main.
